pkg/controller/migration/namespace_migration.go

```diff
@@ -469,18 +470,36 @@ func (m *CoreNamespaceMigration) getNodesToMigrate() []*v1.Node {
 // kubernetes rolling update.
 func (m *CoreNamespaceMigration) waitForCalicoPodsHealthy() error {
 	return wait.PollImmediate(5*time.Second, 1*time.Minute, func() (bool, error) {
-		ksD, ksR, err := m.getNumPodsDesiredAndReady(kubeSystem, nodeDaemonSetName)
+		ksD, ksR, ksMaxUnavailable, err := m.getNumPodsDesiredAndReady(kubeSystem, nodeDaemonSetName)
 		if err != nil {
 			return false, err
 		}
-		csD, csR, err := m.getNumPodsDesiredAndReady(common.CalicoNamespace, nodeDaemonSetName)
+		csD, csR, csMaxUnavailable, err := m.getNumPodsDesiredAndReady(common.CalicoNamespace, nodeDaemonSetName)
 		if err != nil {
 			return false, err
 		}
 
-		// TODO: When we support configuring adjusting the rolling update unavailable,
-		// we should use that configuration here.
-		if ksD == ksR && csD == csR {
+		// MaxUnavailable represents the number of pods that are allowed to be down during an update. We
+		// decrement the number of desired pods by that value
+		if ksMaxUnavailable != nil {
+			n, err := intstr.GetValueFromIntOrPercent(ksMaxUnavailable, int(ksD), false)
+			if err != nil {
+				if (ksD - int32(n)) >= 0 {
+					ksD -= int32(n)
+				}
+			}
+		}
+
+		if csMaxUnavailable != nil {
+			n, err := intstr.GetValueFromIntOrPercent(csMaxUnavailable, int(csD), false)
+			if err != nil {
+				if (csD - int32(n)) >= 0 {
+					csD -= int32(n)
+				}
+			}
+		}
+
+		if ksD <= ksR && csD <= csR {
```

Comment:
I'm not sure if this is what we want. I think this could result in double the number of unavailable pods.
As I was just thinking through this, I think we should use the MaxUnavailable from the calico-system daemonset and make sure that ksR + csR is greater than or equal to that value. I think that would best simulate what a Daemonset RollingUpdate update would do.

Reply:
Yeah, agreed - as it's currently ready I think you could end up with 2*maxUnavailable pods not ready.

We should compare the total number of unready pods vs the total number of desired pods.